src/Model/Resolver/SaveCartItem.php

```diff
@@ -162,12 +162,26 @@ private function prepareOptions(array $options): array
     }
 
     /**
+     * @param string $request
      * @param Product $product
-     * @param array   $options
-     * @return DataObject
+     * @param array $options
+     * @return array
      */
-    private function prepareAddItem(Product $product, array $options): DataObject
-    {
+    private function setOptionsFromBuyRequest(Product $product, array $options) : array {
+        $request = $options['product_option']['buy_request'];
+        $data = json_decode($request, true);
+        $data['product'] = $product->getEntityId();
+        $data['qty'] = $options['quantity'];
+
+        return $data;
+    }
+
+    /**
+     * @param Product $product
+     * @param array $options
+     * @return array
+     */
+    private function setOptionsFromExtensions(Product $product, array $options) : array {
```

Comment:
Will it make sense renaming these two functions? It seems that they don't actually set anything neither on product nor in options, rather just generate some data